crates/fuel-core/src/schema/tx.rs

```diff
@@ -276,16 +277,43 @@ impl TxMutation {
         gas_price: Option<U64>,
     ) -> async_graphql::Result<Vec<DryRunTransactionExecutionStatus>> {
         let block_producer = ctx.data_unchecked::<BlockProducer>();
-        let params = ctx
+        let consensus_params = ctx
             .data_unchecked::<ConsensusProvider>()
             .latest_consensus_params();
+        let block_gas_limit = consensus_params.block_gas_limit();
 
+        let mut max_gas_usable = 0;
         let mut transactions = txs
             .iter()
             .map(|tx| FuelTx::from_bytes(&tx.0))
             .collect::<Result<Vec<FuelTx>, _>>()?;
+        let fee = consensus_params.fee_params();
         for transaction in &mut transactions {
-            transaction.precompute(&params.chain_id())?;
+            match transaction {
+                FuelTx::Script(tx) => {
+                    max_gas_usable += tx.max_gas(consensus_params.gas_costs(), fee)
+                }
+                FuelTx::Create(tx) => {
+                    max_gas_usable += tx.max_gas(consensus_params.gas_costs(), fee)
+                }
+                FuelTx::Mint(_) => {
+                    // Mint transactions doesn't consume gas and so we will add a flat 10% of the block gas limit
+                    max_gas_usable += block_gas_limit / 10;
+                }
+                FuelTx::Upgrade(tx) => {
+                    max_gas_usable += tx.max_gas(consensus_params.gas_costs(), fee)
+                }
+                FuelTx::Upload(tx) => {
+                    max_gas_usable += tx.max_gas(consensus_params.gas_costs(), fee)
+                }
+                FuelTx::Blob(tx) => {
+                    max_gas_usable += tx.max_gas(consensus_params.gas_costs(), fee)
+                }
+            };
+            if max_gas_usable > block_gas_limit {
```

Comment:
I was thinking maybe to add this logic directly into the `dry_run` method of the executor=)

Reply:
In the issue, I understand that you wanted both. I also made the modification in the executor I can remove the graphql part.

Reply:
Ok I was thinking you wanted in the API AND in the executor, will change that :)

Reply:
In the case of dry run we currently ignore `remaining_gas_limit`

Reply:
@xgreenx Could you point me to this ? Because I don't see where is the behavior difference

Reply:
Done.